### ISSUE TITLE
PytorchLightning AimLogger custom contexts

### ISSUE THREAD

**Issue by sbatchelder** [aimhubio/aim#3259]
## 🚀 Feature

The PytorchLightning AimLogger should allow users to define custom logged metric label mappings to aim contexts.
As it stands, `aim.sdk.adapters.pytorch_lightning.AimLogger` can only save metrics to Aim with the "subset" context (for train, val, test groups). The use of customizable prefixes is neat, but the fact that it's only to the "subset" context, that it's only for prefixes, and that the context mapping is defined in three separate init args is limiting. The logger should allow for any number of contexts and label-to-context-value mappings. 

### Motivation

I wanted to log the weighted, macro, and micro metric averaging methods using Aim's context mechanism. I'm using pytorch-lighting and I want to keep the code as lightning-friendly as possible. For instance, without a way to log `val_f1_weighted` to both the `subset:val` and `averaging:weighted` I'm left stuck with multiple aim-logged f1 metrics which should be grouped with contexts, but aren't.

### Alternatives

It's possible to invoke the `trainer.logger.experiment.track(..., context=....)` directly from wherever in the training/validation/test loops/callbacks in a lightning module so that you can log things to Aim with exacting specificity... but I shouldn't have to. It makes the code I write in the lightning modules less portable because it relies on implementation specific calls instead of the preferable built-in lightning logger mechanism.

### Pitch

Update or subclass AimLogger such that it can accept generic metric-label context mappings. Ideally without breaking backwards compatibility and retaining default functionality. 

#### Proposed Usage
```
logger = AimLogger(
    args.repo,
    context_postfixes = dict(
        averaging = {'macro': '_macro',       # for f1, precision, recall
                     'micro': '_micro', 
                     'weighted': '_weighted', 
                     'none': '_perclass'},  
        normalized = {'no': '_summed',        # for confusion matrix
                      'yes': '_normalized'},  
  ))

# then later in the lightning module callback functions
    def validation_step(self, ...):
        ...
        self.log("val_loss", loss)

    def on_validation_epoch_end(self):
        ...
        self.log('val_f1_macro', self.metrics['f1_macro'].compute(), on_epoch=True)
        self.log('val_f1_weighted', self.metrics['f1_weighted'].compute(), on_epoch=True)
```
Note how `self.log(val_loss, ...)` still works because the proposed AimLogger retains the default subset-context defaults, and in fact `self.log(val_f1_weighted, ...)` also benefits from the `val_` subset context mapping.

The proposed AimLogger has a default init value for `context_prefixes` that looks like this:
```
class AimLoggerWithContext(AimLogger):
    def __init__(self,
        ...
        context_prefixes: Optional[Dict] = dict(subset={'train':'train_', 'val':'val_', 'test':'test_'}),
        context_postfixes: Optional[Dict] = dict(),
        ...
```
This maintains the default functionality of the current AimLogger while also granting flexibility. Only users currently instantiating their AimLoggers with `train_metric_prefix`, `val_metric_prefix`, and `test_metric_prefix` args may be affected by my AimLogger refactoring. There are a few ways of addressing this, discussed later. 

### Additional context

I've been using this modified AimLogger for my work, with the above context_postfix settings, and it works great.
![image](https://github.com/user-attachments/assets/152c86f0-5249-45af-bb1f-8487a96c2617)

You may have noticed the `averaging:none`  and confusion-matrix normalization context definitions and wonder whyɁ̦ since you can't log non-point metrics to Aim (or with most lightning loggers) . What I'm doing is using the new AimLogger metric label context mapping mechanism when I create my Aim figures. This lets me extract context from my complex torchmetrics metrics with keys/names the same way I would my point-metric names. 
![image](https://github.com/user-attachments/assets/93ac7084-b144-4c44-bf5a-f9fcb6626d53)
